akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -28,6 +28,16 @@ import akka.util.Timeout
 
   private var messageAdapterRef: OptionVal[ActorRef[Any]] = OptionVal.None
   private var _messageAdapters: List[(Class[_], Any ⇒ T)] = Nil
+  private var _timer: OptionVal[TimerSchedulerImpl[T]] = OptionVal.None
```

Comment:
I wonder if we should create a single `contextLocal` map of some form here to have only a single field cost when not using adapters/timers and live with the indirection cost when using them?

Reply:
I think the cost of map lookup might too high, and using some kind of optimized array based map would defeat the purpose low memory usage